Sources/TSCBasic/Path.swift

```diff
@@ -544,11 +549,13 @@ private struct UNIXPath: Path {
 
     init(normalizingAbsolutePath path: String) {
       #if os(Windows)
-        var buffer: [WCHAR] = Array<WCHAR>(repeating: 0, count: Int(MAX_PATH + 1))
+        var result: PWSTR?
+        defer { LocalFree(result) }
+
         _ = path.withCString(encodedAs: UTF16.self) {
-            PathCanonicalizeW(&buffer, $0)
+            PathAllocCanonicalize($0, ULONG(PATHCCH_ALLOW_LONG_PATHS.rawValue), &result)
         }
-        self.init(string: String(decodingCString: buffer, as: UTF16.self))
+        self.init(string: String(decodingCString: result!, as: UTF16.self))
       #else
```

Comment:
If you want to change this, I think it is better to change this to `PathCchCanonicalize` rather than using `PathAllocCanonicalize`.  I assume that the reason for this change is the long path support?

Reply:
Both seems okay to me, so I picked `PathAllocCanonicalize` for long path support. Deprecating `PathCanonicalizeW` for memory safety.

Reply:
PathCch APIs are recommended.  I don't see what memory safety issues it avoids.  Yes, I get that it is a mutating API, but it is actually safer - it doesn't do additional allocations.